osu.Game/Screens/Play/BeatmapMetadataDisplay.cs

```diff
@@ -197,21 +197,19 @@ protected override void LoadComplete()
                 starRatingDisplay.Show();
             }
             else
-            {
                 starRatingDisplay.Hide();
 
-                starDifficulty.ValueChanged += d =>
-                {
-                    Debug.Assert(d.NewValue != null);
+            starDifficulty.ValueChanged += d =>
```

Comment:
Is there much reason to keep the above `Show` logic rather than use `BindValueChanged(...,true)`? Second is preferable I think.

Reply:
Haven't done that because the initial logic above shows star rating display immediately, but if the calculation takes longer, it fades in smoothly once it's ready instead. Could merge together and use `d.NewValue == d.OldValue` to decide the transition, but not sure that reads better.

Reply:
okay, let's go with it then. didn't see the initial `Alpha=0`.